app/gui2/src/components/visualizations/TableVisualization.vue

```diff
@@ -341,6 +366,16 @@ watchEffect(() => {
     page.value = newPageLimit
   }
 
+  if (rowData.length) {
+    const headers = Object.keys(rowData[0] as object)
+    let obj = {}
+    headers.forEach((header) => {
+      const needsGrouping = rowData.some((row: any) => row[header] > 9999)
+      obj = { ...obj, [header]: needsGrouping }
+    })
+    dataGroupingMap.value = obj
```

Comment:
I'd prefer this to be implemented using a `computed` instead of updating a ref manually, but I guess this is the right thing to do in this case. Using `computed` here would indeed require more significant refactor of the above logic.

Since you are updating the whole `dataGroupingMap` object at once, use `shallowRef` instead of `ref` to avoid creating unnecessary read dependencies on individual fields, since the object is replaced as a whole anyway.